Board/Board.cpp

```diff
@@ -0,0 +1,128 @@
+//
+// Created by Yam on 11/06/2023.
+//
+
+#define DETAILS_DELIMITER ','
+
+#include <fstream>
+#include <sstream>
+#include <typeindex>
+#include <map>
+
+#include "Board.h"
+#include "Piece/Pieces/Pawn/Pawn.h"
+#include "Piece/Pieces/Rook/Rook.h"
+#include "Piece/Pieces/Knight/Knight.h"
+#include "Piece/Pieces/Bishop/Bishop.h"
+#include "Piece/Pieces/Queen/Queen.h"
+#include "Piece/Pieces/King/King.h"
+
+
+Board::Board()
+{
+    Color current_position_color = kWhite;
+    for (int i = 0; i < BOARD_WIDTH; ++i)
+    {
+        for (int j = 0; j < BOARD_LENGTH; ++j)
+        {
+            board[i][j].initializePosition(i, j, current_position_color, nullptr);
```

Comment:
After change to pointer, remove this function and initialize inside the constructor